examples/ffmpeg-react/src/App.tsx

```diff
@@ -0,0 +1,296 @@
+
+    const [outputVideo, setOutputVideo] = useState<VideoProps>({
+        preview: false,
+        fileSrc: "",
+        file: null,
+    });
+
+    const [processingStatus, setProcessingStatus] = useState<PROCESSING_STATUS>(
+        PROCESSING_STATUS.NOT_STARTED,
+    );
+
+    const previewVideoRef = useRef<HTMLVideoElement>(null);
+    const outputVideoRef = useRef<HTMLVideoElement>(null);
+
+    const [fileU8Arr, setFileU8Arr] = useState<Uint8Array | null>(null);
+
+    const onDrop = useCallback((acceptedFiles: File[]) => {
+        const file = acceptedFiles[0];
+        console.log(file);
+        setUserVideo({
+            preview: true,
+            fileSrc: URL.createObjectURL(file),
+            file,
+        });
+    }, []);
+
+    const { getRootProps, getInputProps } = useDropzone({
+        maxFiles: 1,
+        accept: {
+            "video/*": [],
+        },
+        onDrop,
+    });
+
+    useEffect(() => {
+        if (!userVideo.fileSrc || !previewVideoRef.current) return;
+        previewVideoRef.current?.load();
+    }, [userVideo.fileSrc]);
+
+    useEffect(() => {
+        if (!outputVideo.fileSrc || !outputVideoRef.current) return;
+        outputVideoRef.current?.load();
+    }, [outputVideo.fileSrc]);
+
+    useEffect(() => {
+        (async () => {
+            if (loggerInitialized) return;
+            loggerInitialized = true;
+            await init();
+            // initializeLogger(
+            //     "info,wasmer_wasix=debug,wasmer_wasix::syscalls=debug,wasmer_js=debug",
+            // );
+        })();
+    }, []);
+
+    useEffect(() => {
+        if (!userVideo.file) {
+            return;
+        }
+        const reader = new FileReader();
+
+        reader.onload = function (e: ProgressEvent<FileReader>) {
+            if (!e.target) return;
+            const arrayBuffer = e.target.result as ArrayBuffer;
+            const u8arr = new Uint8Array(arrayBuffer);
+
+            setFileU8Arr(u8arr);
+        };
+
+        reader.readAsArrayBuffer(userVideo.file);
+    }, [userVideo.file]);
+
+    useEffect(() => {
+        if (processingStatus === PROCESSING_STATUS.FAILED) {
+            setTimeout(() => {
+                setProcessingStatus(PROCESSING_STATUS.NOT_STARTED);
+            }, 4000);
+        }
+    }, [processingStatus]);
+    const runFfmpegProcessing = async () => {
+        if (!fileU8Arr) return;
+
+        const runtime = new Runtime({
+            poolSize: 64,
+        });
+
+        setProcessingStatus(PROCESSING_STATUS.RUNNING);
+
+        const tmp = new Directory();
+        await tmp.writeFile("input.mp4", fileU8Arr);
+        const pkg = await Wasmer.fromRegistry("wasmer/ffmpeg", runtime);
+
+        if (!pkg.entrypoint) return;
+
+        const instance = await pkg.entrypoint!.run({
+            args: [
+                "-i",
+                "/videos/input.mp4",
+                "-vf",
+                "format=gray",
+                "/videos/output.mp4",
+            ],
```

Comment:
Can we make this input actually modifiable by the user?

Reply:
Let's leave the UI/UX for later.